pytket/binders/circuit/boxes.cpp

```diff
@@ -47,6 +53,18 @@ std::map<py::tuple, T2> cast_keys_to_tuples(
   }
   return outmap;
 }
+// Cast std::map<T1, T2> to std::vector<std::pair<T1, T2>>, useful when T1 or T2
+// is a vector because list is not hashable in python
```

Comment:
The way it's written is correct (except that I think its actually ok if T2 is a vector, Ill change that), although maybe confusing. The problem is that `list` objects aren't hashable. pybind11 will translate `map<vector, T>` to Dict[List, pyT]. But actually creating such an object is not possible in python because the keys of dicts have to be hashable. In this case the keys would be lists, which aren't. As a workaround, we were using tuples instead of lists in the test code, but that makes mypy angry

Reply:
Ah OK